src/test/java/com/edmunds/tools/databricks/maven/PrepareLibraryResourcesTest.java

```diff
@@ -1,36 +1,55 @@
+/*
+ *    Copyright 2018 Edmunds.com, Inc.
+ *
+ *        Licensed under the Apache License, Version 2.0 (the "License");
+ *        you may not use this file except in compliance with the License.
+ *        You may obtain a copy of the License at
+ *
+ *            http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *        Unless required by applicable law or agreed to in writing, software
+ *        distributed under the License is distributed on an "AS IS" BASIS,
+ *        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *        See the License for the specific language governing permissions and
+ *        limitations under the License.
+ */
+
 package com.edmunds.tools.databricks.maven;
 
-import org.testng.annotations.BeforeMethod;
+import org.apache.maven.plugin.MojoExecutionException;
 import org.testng.annotations.Test;
 
-
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 
-public class PrepareLibraryResourcesTest extends BaseDatabricksMojoTest {
-
-    private PrepareLibraryResources underTest = new PrepareLibraryResources();
+//Requires building beforehand??
+public class PrepareLibraryResourcesTest extends DatabricksMavenPluginTestHarness {
 
-    @BeforeMethod
-    public void init() throws Exception {
-        super.init();
+    private final String GOAL = "prepare-library-resources";
 
-        underTest.setDatabricksServiceFactory(databricksServiceFactory);
-        underTest.setProject(project);
-        underTest.setVersion("1.0");
-        underTest.setDbfsRoot("s3://bucket-name");
+    @Test
+    public void testCreateArtifactPath_default() throws Exception {
+        PrepareLibraryResources underTest = (PrepareLibraryResources) getNoOverridesMojo(GOAL);
+        assertThat(underTest.createArtifactPath(), is("s3://my-bucket/artifacts/unit-test-group" +
+                "/unit-test-artifact/1.0.0-SNAPSHOT/unit-test-artifact-1.0.0-SNAPSHOT.jar"));
+        //TODO actually test the execute here
+        underTest.execute();
     }
 
     @Test
-    public void testCreateArtifactPath_default() throws Exception {
-        assertThat(underTest.createArtifactPath(), is("s3://bucket-name/artifacts/com.edmunds" +
-                ".test/mycoolartifact/1.0/mycoolartifact-1.0.jar"));
+    public void testCreateArtifactPath_failsWhenMissingMandatoryFields() throws Exception {
+        PrepareLibraryResources underTest = (PrepareLibraryResources) getMissingMandatoryMojo(GOAL);
+        try {
+            underTest.execute();
+        } catch (MojoExecutionException e) {
+            return;
```

Comment:
Same thing here, I think you want to expect the exception in the method annotation instead of this return + fail pattern.

Reply:
the annotation approach doesn't work because of MojoExecutionExceptions specifically. It won't work as far as I can tell during a maven run of the tests. I have a comment from when we tried to do this in a previous commit.

Reply:
Hmm, that's odd. What about that exception makes it not work?